internal/sessions/state.go

```diff
@@ -182,3 +183,24 @@ func (s *State) accessTokenHash() string {
 	}
 	return fmt.Sprintf("%x", hash)
 }
+
+// UnmarshalJSON parses the JSON-encoded session state.
+// TODO(BDD): remove in v0.8.0
+func (s *State) UnmarshalJSON(b []byte) error {
+	type Alias State
+	t := &struct {
+		*Alias
+		OldToken *oauth2.Token `json:"access_token,omitempty"` // < v0.5.0
+	}{
+		Alias: (*Alias)(s),
+	}
+	if err := json.Unmarshal(b, &t); err != nil {
+		return err
+	}
+	if t.AccessToken == nil {
+		t.AccessToken = t.OldToken
+	}
+	*s = *(*State)(t.Alias)
```

Comment:
Seems fine though a bit opaque.  

Soft suggestion - Would it be possible to put the old field in `State`, and just worry about copying `OldToken` into `AccessToken` (if present) here?  I'm not sure if there's a downside to that approach but the code would be a lot clearer here.

Reply:
I explored a few different options.

### Using a getter.

> Soft suggestion - Would it be possible to put the old field in State, and just worry about copying OldToken into AccessToken (if present) here? I'm not sure if there's a downside to that approach but the code would be a lot clearer here.

This was my first thought, and approach. Would have resulted in... hundreds loc changes. Broke tests, and touched so many unrelated areas of the code base I am reticent to go this approach for a hotfix.

### Using a custom type for access token only.

This was my 2nd approach. The problem I found here was I still needed to make a bunch of unrelated code changes and higher level type casts. 

```go
type State struct {
	AccessToken   *oauthToken 
}

type oauthToken struct {
	*oauth2.Token
}
func (oa *oauthToken) UnmarshalJSON(b []byte) error {
	t := &struct {
		NewToken *oauth2.Token `json:"act,omitempty"`          // > v0.5.0
		OldToken *oauth2.Token `json:"access_token,omitempty"` // < v0.5.0
	}{}
	if err := json.Unmarshal(b, &t); err != nil {
		return err
	}
	oa = &oauthToken{t.NewToken}
	if oa == nil {
		oa = &oauthToken{t.OldToken}
	}
	return nil
}```

Reply:
Ouch.  Well given it's due to be removed, best to keep it contained.

Reply:
I think approach 1. is probably the best, in general, but for now.. I think this will work.